### DISCUSSION TITLE
How to shard tests across a GitHub matrix

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [vitest-dev/vitest#4752] [Q&A] [answered]
I'm basically looking for guidance like this: https://playwright.dev/docs/test-sharding#github-actions-example

Two things:

1. Vitest should support this if it does not, I think
2. If it is supported, it seems the documentation about it needs to be improved/written.

Any help appreciated thanks!

**Comment by AriPerkkio** [comment, marked as answer]
Comparing to the Playwright's documentation it should be as simple as:

```diff
-  playwright-tests:
+  vitest-tests:
    timeout-minutes: 60
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        shard: [1/4, 2/4, 3/4, 4/4]

...

- - name: Run Playwright tests
-   run: npx playwright test --shard ${{ matrix.shard }}
+ - name: Run Vitest tests
+   run: npx vitest --shard ${{ matrix.shard }}
```

Any recommendations what to change here? https://vitest.dev/guide/cli.html#shard

**Reply by jasonkuhrt** [reply]
Thanks!

> Any recommendations what to change here? [vitest.dev/guide/cli.html#shard](https://vitest.dev/guide/cli.html#shard)

1. Make a guide section about sharding
2. In that guide add an example using GitHub actions
3. Mention case of using coverage, too


The problem with the current documentation is less what is there and what is not there. It relies on users knowing that sharding is a concept through the CLI, and that it is called sharding. It is not discoverable enough for people that don't even have that orientation of Vitest or terms. In general I think that's one thing that differentiates guide sections from api ones.

**Reply by AriPerkkio** [reply]
The documentation could indeed be improved. I had no idea where `--shard` feature would be described so I simply used search and the page above came up in first result.

**Comment by jasonkuhrt** [comment]
@AriPerkkio I've opened a new question for an aspect that is not resolved here but wasn't in my original question so... https://github.com/vitest-dev/vitest/discussions/4755
